src/persist/src/postgres.rs

```diff
@@ -188,19 +205,36 @@ impl PostgresConsensus {
         }
         tx.batch_execute(SCHEMA).await?;
         tx.commit().await?;
-        Ok(PostgresConsensus { pool })
+        Ok(PostgresConsensus {
+            pool,
+            metrics: config.metrics,
+        })
     }
 
     /// Drops and recreates the `consensus` table in Postgres
     ///
     /// ONLY FOR TESTING
     pub async fn drop_and_recreate(&self) -> Result<(), ExternalError> {
         // this could be a TRUNCATE if we're confident the db won't reuse any state
-        let client = self.pool.get().await?;
+        let client = self.get_connection().await?;
         client.execute("DROP TABLE consensus", &[]).await?;
         client.execute(SCHEMA, &[]).await?;
         Ok(())
     }
+
+    async fn get_connection(&self) -> Result<Object, PoolError> {
+        let start = Instant::now();
+        let res = self.pool.get().await;
+        self.metrics
+            .connpool_acquire_time_seconds
+            .inc_by(start.elapsed().as_secs_f64());
+        self.metrics.connpool_acquires.inc();
+        // note that getting the pool size here requires briefly locking the pool
```

Comment:
after looking at it very briefly, I think maybe the reason there isn't a delete hook is that the pool doesn't actually delete. thoughts on running this logic only in the `post_create` hook? (do we even have access to what we need to do that in the post_create hook?

Reply:
I tried that too and sadly, no. we'd need access to the pool, but the hooks are defined while creating the pool, so there's nothing to refer to. there also isn't a way to add a hook after creation

Reply:
👍